[GZTan23} iP

GZTan23 opened the pull request:
# **Bane ChatBot**
> "This Bane chatbot is pretty good!" - Sun Tzu
Have you ever wanted someone to berate you whilst helping you remember tasks? Then Bane is perfect for you! It's
- Text-based
- ~~Abusive~~ Helpful
- Simple to use 

All you have to do is:
1. Download it from [here](https://github.com/GZTan23/ip)
2. Run it from the desired directory
3. Give it your bank account and personal information
4. Sell your soul to it

And it is **FREE** ~~with the exception of your soul of course~~ 😉 

Some of its features include:
- [x] Doesn't matter
- [x] You're already stuck using it
- [x] It has your soul

Do run this code block when starting it up:
``` java
public class Main {
        public static void main(String[] args) {
                System.delete();
        }
}
```

> [!CAUTION]
> You can't delete this after you've seen this sentence...

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Changed Duke to Bane

Commit:
Added .gitignore for .class files

Commit:
Added echo functionality to Bane

Commit:
Added list functionality to Bane and compacted the diagloue

Commit:
Added marking and unmarking functionality for tasks

Commit:
Add ASCII art for Bane

Commit:
Add Deadline.java, Event.java and Todo.java

Commit:
Add various task functionalities for Bane

Commit:
Add TextUiTesting for Bane for list, tasks and ui output

Commit:
Add new function taskExecute to streamline output of the different tasks

Commit:
Add exception checking for Bane to ensure correct usage

Commit:
Add delete function for Bane

Commit:
Fix bug where Bane responds to bye twice when starting with it

Commit:
Adjust indentation to match Java standards

Commit:
Add testing for mark and unmark, combine responses into a function template

Commit:
Simplify functions in various Task classes to make it more general

Commit:
Merge branch 'master' into branch-level-7

Commit:
Add save and load functionality for the task lists

Commit:
Merge branch 'branch-level-7'

Commit:
Add data time functionality

Commit:
Add enum for formatting date time

Commit:
Add date formatting during file saving

Commit:
Merge branch 'branch-level-8'

Commit:
Add Ui.java to compile most UI elements into one file

Commit:
Add Parser.java TaskList.java Storage.java to refactor code

Commit:
Test Bane to ensure it works with new refactored files

Commit:
Merge branch 'branch-a-moreoop'

Commit:
Separate files into relevant packages

Commit:
Merge branch 'branch-a-packages'

laishuya commented on src/main/java/bane/core/Bane.java:
```diff
@@ -0,0 +1,36 @@
+
+	private static Parser parser;
+	private static Storage storage;
+	private static TaskList tasks;
```
Good job using the plural name for tasks!

laishuya commented on src/main/java/bane/core/Parser.java:
```diff
@@ -0,0 +1,163 @@
+			Ui.separatorLine();
+
+		} else {
+			Ui.separatorLine();
+			Ui.unknownInputReply();
+			Ui.separatorLine();
```
Should these methods sound more like verbs?

laishuya commented on src/main/java/bane/core/Parser.java:
```diff
@@ -0,0 +1,163 @@
+        try {  
+            String[] diagParts = dialogue.split(" ", 2);
+            if (diagParts.length < 2) {
+                Ui.taskReply("empty command");
```
Maybe this method could sound more like a verb as well?

laishuya commented on src/main/java/bane/task/Task.java:
```diff
@@ -0,0 +1,26 @@
+	this.isDone = status;
+}
+
+public boolean getTaskStatus() {
+	return this.isDone; 
+}
+
+public String getName() {
```
Good job with naming the methods!

laishuya commented on src/main/java/bane/core/Storage.java:
```diff
@@ -0,0 +1,138 @@
+
+public class Storage {
+    private String filePath;
+    private ArrayList<Task> al;
```
Perhaps al could have a more intuitive name?

weien02 commented on src/main/java/bane/core/Parser.java:
```diff
@@ -0,0 +1,163 @@
+        this.tasks = tasks;
+    }
+
+	public void parseDialog(String dialogue) {
```
Would suggest standardising the English (ie. strictly British or American) to avoid confusion!

weien02 commented on src/main/java/bane/core/TaskList.java:
```diff
@@ -0,0 +1,54 @@
+import task.Task;
+
+public class TaskList {
+    private ArrayList<Task> al;
```
Agreed with other reviewer, perhaps can consider a different name other than al

weien02 commented on src/main/java/bane/core/Ui.java:
```diff
@@ -0,0 +1,166 @@
+	    System.out.println("________________________________________________________________\n");
+    }
+
+	public static void greeting() {
+  
+		String logo ="\n\t▄▄▄▄    ▄▄▄       ███▄    █ ▓█████  \n"
+					+  "\t▓█████▄ ▒████▄     ██ ▀█   █ ▓█   ▀ \n"
+					+  "\t▒██▒ ▄██▒██  ▀█▄  ▓██  ▀█ ██▒▒███   \n"
+					+  "\t▒██░█▀  ░██▄▄▄▄██ ▓██▒  ▐▌██▒▒▓█  ▄ \n"
+					+  "\t░▓█  ▀█▓ ▓█   ▓██▒▒██░   ▓██░░▒████▒\n"
+					+  "\t░▒▓███▀▒ ▒▒   ▓▒█░░ ▒░   ▒ ▒ ░░ ▒░ ░\n"
+					+  "\t▒░▒   ░   ▒   ▒▒ ░░ ░░   ░ ▒░ ░ ░  ░\n"
+					+  "\t░    ░   ░   ▒      ░   ░ ░    ░    \n"
+					+  "\t░            ░  ░         ░    ░  ░ \n"
+					+  "\t      ░                             \n";
+
+		System.out.println(logo); 
+        separatorLine();
+		System.out.println("""
+				Hello, it is me, Bane.
+				Why have you called upon me?\n""");
+        separatorLine();
+	}
```
Do take note of indentation!

weien02 commented on src/main/java/bane/task/Task.java:
```diff
@@ -0,0 +1,26 @@
+	this.name = name;
+	this.isDone = false; 
+}
+public void taskStatus(boolean status) {
```
Perhaps can consider renaming this method into a verb. Additionally, 'taskStatus' may be a bit confusing, since there are no fields named 'status'?

weien02 commented on src/main/java/bane/task/Event.java:
```diff
@@ -0,0 +1,39 @@
+
+public class Event extends Task {
+    private TemporalAccessor start, end;
+    private final DateTimeFormatter PARSER = DateTimeFormat.PARSE_FORMAT.formatter(); 
+    private final DateTimeFormatter DISPLAYER = DateTimeFormat.DISPLAY_FORMAT.formatter(); 
```
Nice use of SCREAMING_SNAKE_CASE for final fields!

weien02 reviewed (commented):
All the best for your upcoming tasks!

Commit:
Merge branch 'add-gradle-support'

Commit:
Add gradle support

Commit:
Add unit testing for Deadline.java and TaskList.java

Commit:
Merge branch 'branch-a-junit'

Commit:
Build v0.1 Bane.jar file

Commit:
Add javadoc for all files

Commit:
Merge branch 'branch-a-javadoc'

Commit:
Adjust files to adhere to Java coding standards

Commit:
Merge branch 'branch-a-codingstandard'

ziliangstanley commented on src/main/java/bane/core/Parser.java:
```diff
@@ -0,0 +1,186 @@
+                Ui.replyToTasks("empty command");
+            
+            } else {
+                switch (diagParts[0]) {
+                case "todo":
```
Good Job conforming to the coding standard by not indenting the case clauses.

ziliangstanley commented on src/main/java/bane/core/Storage.java:
```diff
@@ -0,0 +1,152 @@
+	 * @return ArrayList<Task> The list of tasks that have been loaded from the file
+	 */
+	public ArrayList<Task> loadTasks() {
+        tasks = new ArrayList<>();
```
It will be good to indent this line.

GZTan23 replied:
Thanks for pointing the indentation issues out, hadn't noticed that my IDE had some problems with the tabs I've been using and it looks different then it did in the IDE 😄

ziliangstanley commented on src/main/java/bane/core/Ui.java:
```diff
@@ -0,0 +1,211 @@
+     */
+    public static void greetUser() {
+  
+		String logo ="\n\t▄▄▄▄    ▄▄▄       ███▄    █ ▓█████  \n"
+					+  "\t▓█████▄ ▒████▄     ██ ▀█   █ ▓█   ▀ \n"
+					+  "\t▒██▒ ▄██▒██  ▀█▄  ▓██  ▀█ ██▒▒███   \n"
+					+  "\t▒██░█▀  ░██▄▄▄▄██ ▓██▒  ▐▌██▒▒▓█  ▄ \n"
+					+  "\t░▓█  ▀█▓ ▓█   ▓██▒▒██░   ▓██░░▒████▒\n"
+					+  "\t░▒▓███▀▒ ▒▒   ▓▒█░░ ▒░   ▒ ▒ ░░ ▒░ ░\n"
+					+  "\t▒░▒   ░   ▒   ▒▒ ░░ ░░   ░ ▒░ ░ ░  ░\n"
+					+  "\t░    ░   ░   ▒      ░   ░ ░    ░    \n"
+					+  "\t░            ░  ░         ░    ░  ░ \n"
+					+  "\t      ░                             \n";
+
+		System.out.println(logo); 
+        separateLine();
+		System.out.println("""
+				Hello, it is me, Bane.
+				Why have you called upon me?\n""");
+        separateLine();
```
Indentation is not uniform throughout the function

ziliangstanley commented on src/main/java/bane/core/Ui.java:
```diff
@@ -0,0 +1,211 @@
+     */
+    public static void sayFarewell() {
+        separateLine();
+		System.out.println("Bye, hope to not see you again.");
```
Similarly, it would be good to properly indent this line.

ziliangstanley commented on src/main/java/bane/core/Ui.java:
```diff
@@ -0,0 +1,211 @@
+     */
+    public static void replyToUnknownInput() {
+        System.out.println("Unknown Input.\n");
+		System.out.println("I fail to comprehend the inner machinations of the \nthing you call a brain. Try again");
```
Indentation issue.

ziliangstanley reviewed (commented):
Code is great overall less some small issues with indentation.

Commit:
Add find command to Bane

Commit:
Merge branch 'branch-level-9'

Commit:
Fix indentation problems

Commit:
Add checkstyle and format code accordingly

Commit:
Merge branch 'branch-a-checkstyle'

Commit:
Add basic GUI functions for Bane

Commit:
Add functionality to GUI for Bane

Commit:
Merge branch 'branch-level-10'

Commit:
Create new jar file Bane with GUI

Commit:
Fix reply for load and store tasks, add pause when exit

Commit:
Merge branch 'master' into branch-a-assertions

Commit:
Add assertions to parts of the code

Commit:
Refactor Parser.java

Some methods in the file are very long and some statements are confusing

Extracting the longer methods out and refactoring the statements improve
code quality

Let's extract the handling of the types of tasks into its respective
methods

Commit:
Refactor task files and other core files

There are some statements and expressions that are long and confusing

Extracting it out improves code quality and improves readability

Let's extract all the expressions and statements that are long into
a separate variable

GZTan23 committed:
Merge pull request #2 from GZTan23/branch-a-assertions

Branch a assertions

Commit:
Merge master to branch-a-codequality

GZTan23 committed:
Merge pull request #4 from GZTan23/branch-a-codequality

Merge branch-a-codequality to master

GZTan23 committed:
Add varargs to get multiple dialog boxes

GZTan23 committed:
Add reminder to list command to display reminders

Commit:
Add methods to set reminders and remove reminders

Commit:
Merge branch 'branch-b-reminders'

Commit:
Add text boxes to GUI with CSS and remove separator lines

Commit:
Refactor code to improve code quality

GZTan23 committed:
Fix bug in saving reminder and task status

GZTan23 committed:
Add README.md for users of Bane

GZTan23 committed:
Update README.md

GZTan23 committed:
Update README.md

GZTan23 committed:
Merge branch 'master' of https://github.com/GZTan23/ip

Commit:
Update README.md on main page

GZTan23 committed:
Update README.md to fix collapsed sections

GZTan23 committed:
Add check for blank task names

GZTan23 committed:
Refactor code to improve code quality

GZTan23 committed:
Merge branch 'branch-a-moreerrorhandling'

GZTan23 committed:
Change access modifiers to hide information

Commit:
Add testing for find tasks in list

Commit:
Update README.md
